src/compiler/watchPublic.ts

```diff
@@ -118,11 +119,8 @@ export function readBuilderProgram(compilerOptions: CompilerOptions, host: ReadB
     return createBuilderProgramUsingProgramBuildInfo(buildInfo, buildInfoPath, host);
 }
 
-export function createIncrementalCompilerHost(options: CompilerOptions, system?: System): CompilerHost;
-/** @internal */
-export function createIncrementalCompilerHost(options: CompilerOptions, system?: System, skipNonSemanticJSDocParsing?: boolean): CompilerHost;
-export function createIncrementalCompilerHost(options: CompilerOptions, system = sys, skipNonSemanticJSDocParsing?: boolean): CompilerHost {
-    const host = createCompilerHostWorker(options, /*setParentNodes*/ undefined, skipNonSemanticJSDocParsing, system);
+export function createIncrementalCompilerHost(options: CompilerOptions, system = sys, jsDocParsingMode?: JSDocParsingMode): CompilerHost {

```

Comment:
Again i dont think we should be exposing this externally unless we find users for the API.

Reply:
My same commment about `createCompilerHost` applies here; an example user is rushstack's `heft` which is another user I've been meaning to talk to about this: https://github.com/octogonz/rushstack/blob/15680b899a06c614c32088b02940bf676596bcf2/apps/heft/src/plugins/TypeScriptPlugin/TypeScriptBuilder.ts#L1079

Reply:
(they use both `createCompilerHost` and the incremental version)

Reply:
I'm also kinda shocked that this function doesn't also provide `setParentNodes`.

Reply:
In fact, I'm inclined to add them just to make sure we end up with the params in the same order...

Reply:
My dreams here have been dashed a bit because `createCompilerHost` doesn't take `system` but `createIncrementalCompilerHost` does. I feel like that should be fixed too.

Reply:
I would keep createCompilerHost as is and add jsdoc thing like you have. As i said there this doesnt make me happy but its ok.